hugegraph-core/src/main/java/com/baidu/hugegraph/schema/builder/IndexLabelBuilder.java

```diff
@@ -554,4 +618,45 @@ private boolean oneNumericField() {
         DataType dataType = propertyKey.dataType();
         return dataType.isNumber() || dataType.isDate();
     }
+
+    private void checkStableVars() {
+        if (this.baseType != HugeType.SYS_SCHEMA) {
+            throw new NotAllowException("Not allowed to update base type " +
+                                        "for index label '%s'", this.name);
+        }
+        if (this.baseValue != null) {
+            throw new NotAllowException("Not allowed to update base value " +
+                                        "for index label '%s'", this.name);
+        }
+        if (this.indexType != IndexType.SECONDARY) {
```

Comment:
why must be SECONDARY

Reply:
SECONDARY is the default value, it means that not allowed to update index type

Reply:
let null as the default value